india_compliance/gst_india/utils/test_e_waybill.py

```diff
@@ -0,0 +1,671 @@
+import random
+import re
+
+import responses
+from responses import matchers
+
+import frappe
+from frappe.tests.utils import FrappeTestCase, change_settings
+from frappe.utils import add_to_date, get_datetime, getdate, now_datetime, today
+from frappe.utils.data import format_date
+
+from india_compliance.gst_india.api_classes.base import BASE_URL
+from india_compliance.gst_india.utils import load_doc
+from india_compliance.gst_india.utils.e_waybill import (
+    EWaybillData,
+    cancel_e_waybill,
+    fetch_e_waybill_data,
+    generate_e_waybill,
+    update_transporter,
+    update_vehicle_info,
+)
+from india_compliance.gst_india.utils.tests import (
+    append_item,
+    create_purchase_invoice,
+    create_sales_invoice,
+)
+
+DATETIME_FORMAT = "%d/%m/%Y %I:%M:%S %p"
+DATE_FORMAT = "dd/mm/yyyy"
+
+
+class TestEWaybill(FrappeTestCase):
+    @classmethod
+    def setUpClass(cls):
+        frappe.db.set_single_value(
+            "GST Settings",
+            {
+                "enable_api": 1,
+                "enable_e_invoice": 0,
+                "auto_generate_e_invoice": 0,
+                "enable_e_waybill": 1,
+                "fetch_e_waybill_data": 0,
+                "auto_generate_e_waybill": 0,
+                "attach_e_waybill_print": 0,
+            },
+        )
+
+        cls.e_waybill_test_data = frappe._dict(
+            frappe.get_file_json(
+                frappe.get_app_path(
+                    "india_compliance", "gst_india", "data", "test_e_waybill.json"
+                )
+            )
+        )
+
+        cls.sales_invoice = _create_sales_invoice(cls.e_waybill_test_data)
+
+    @classmethod
+    def tearDownClass(cls):
```

Comment:
May not be required as the following is done in setUpClass of FrappeTestCase
`cls.addClassCleanup(_rollback_db)`

You may need to do `super().setUpClass()`